gateway-service-knoxtoken/src/main/java/org/apache/knox/gateway/service/knoxtoken/OAuthResource.java

```diff
@@ -90,6 +91,7 @@ public Response getAuthenticationToken() {
             map.put(ISSUED_TOKEN_TYPE, ISSUED_TOKEN_TYPE_ACCESS_TOKEN_VALUE);
             // let's use the passcode as the refresh token
             map.put(REFRESH_TOKEN, passcode);
+            map.put(LIFETIME, getTokenLifetimeInSeconds());
```

Comment:
If I am reading this correctly, then returning the actual value in the response as lifetime_secs would be unexpected by OAuth flow consumers/clients. They will be looking for expires_in.

I think what we need to do is calculate both the existing expires_in which millis since epoch as well as this OAuth expires_in which is lifetime_secs and return one in the response and store the other in the token metadata for reaping and monitoring, etc.